playground/infrastructure/helper.py

```diff
@@ -44,29 +57,213 @@ def find_examples(work_dir: str) -> List[Example]:
         categories:
             - category-1
             - category-2
+    If some example contain beam tag with incorrect format raise an error.
 
     Args:
         work_dir: directory where to search examples.
+        supported_categories: list of supported categories.
 
     Returns:
         List of Examples.
     """
-    # TODO [BEAM-13135] Implement
-    pass
+    has_error = False
+    examples = []
+    for root, _, files in os.walk(work_dir):
+        for filename in files:
+            filepath = os.path.join(root, filename)
+            has_error = has_error or _check_file(examples, filename, filepath, supported_categories)
```

Comment:
This way you won't check the rest of the files after some of them failed because of condition optimizations – if `has_error` is already `True` it won't got through `_check_file`
```suggestion
            result = _check_file(examples, filename, filepath, supported_categories)
            if not has_error and result:
                has_error = result
```

Reply:
Changed.